drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1801 @@
+end
+
+-------------------------
+-- Max PIN Code Length --
+------------------------- 
+local function max_pin_code_len_handler(driver, device, ib, response)
+  log.info_with({hub_logs=true}, string.format("!!!!!!!!!!!!!!! max_pin_code_len_handler: %d !!!!!!!!!!!!!", ib.data.value)) -- needs to remove
+  device:emit_event(capabilities.lockCredentials.maxPinCodeLen(ib.data.value, {visibility = {displayed = false}}))
+end
+
+--------------------------------------
+-- Require PIN For Remote Operation --
+--------------------------------------
+--- If a device needs a cota credential this function attempts to set the credential
+--- at the index provided. The set_credential_response_handler handles all failures
+--- and retries with the appropriate index when necessary.
+local function set_cota_credential(device, credential_index)
+  local eps = device:get_endpoints(DoorLock.ID)
+  local cota_cred = device:get_field(lock_utils.COTA_CRED)
+  if cota_cred == nil then
+    -- Shouldn't happen but defensive to try to figure out if we need the cota cred and set it.
+    device:send(DoorLock.attributes.RequirePINforRemoteOperation:read(device, #eps > 0 and eps[1] or 1))
+    device.thread:call_with_delay(2, function(t) set_cota_credential(device, credential_index) end)
+  elseif not cota_cred then
+    device.log.debug("Device does not require PIN for remote operation. Not setting COTA credential")
+    return
+  end
+
+  if device:get_field(lock_utils.SET_CREDENTIAL) ~= nil then
+    device.log.debug("delaying setting COTA credential since a credential is currently being set")
+    device.thread:call_with_delay(2, function(t)
+      set_cota_credential(device, credential_index)
+    end)
+    return
+  end
+
+  device:set_field(lock_utils.COTA_CRED_INDEX, credential_index, {persist = true})
+  local credential = {credential_type = DoorLock.types.DlCredentialType.PIN, credential_index = credential_index}
+  -- Set the credential to a code
+  device:set_field(lock_utils.COMMAND_NAME, "addCredential")
+  device:set_field(lock_utils.CRED_INDEX, credential_index)
+  device:set_field(lock_utils.SET_CREDENTIAL, credential_index)
+  device:set_field(lock_utils.USER_TYPE, "adminMember")
+  device.log.info(string.format("Attempting to set COTA credential at index %s", credential_index))
+  device:send(DoorLock.server.commands.SetCredential(
+    device,
+    #eps > 0 and eps[1] or 1,
+    DoorLock.types.DlDataOperationType.ADD,
+    credential,
+    device:get_field(lock_utils.COTA_CRED),
+    nil, -- nil user_index creates a new user
+    DoorLock.types.DlUserStatus.OCCUPIED_ENABLED,
+    DoorLock.types.DlUserType.REMOTE_ONLY_USER
+  ))
+end
+
+local function generate_cota_cred_for_device(device)
+  local len = device:get_latest_state("main", capabilities.lockCredentials.ID, capabilities.lockCredentials.maxPinCodeLen.NAME) or 6
+  local cred_data = math.floor(math.random() * (10 ^ len))
```

Comment:
We should seed the PRNG with `math.randomseed(os.time())` like the other drivers that use `math.random()` although it looks like that wasn't done in the original lock driver either.

Reply:
random is pre-seeded at startup, no need to reseed here

Reply:
I'm observing that sometimes math.random() returns the same value on subsequent driver runs. I'll file that under a different ticket though.